Palace/Settings/AccountList/TPPAccountListDataSource.swift

```diff
@@ -0,0 +1,60 @@
+import Foundation
+
+protocol DataSourceDelegate: AnyObject {
+  func refresh()
+}
+
+class TPPAccountListDataSource: NSObject {
+
+  weak var delegate: DataSourceDelegate?
+  var title: String = NSLocalizedString("Pick Your Library", comment: "Title that also informs the user that they should choose a library from the list.")
+  
+  private var accounts: [Account]!
+  private var nationalAccounts: [Account]!
+  
+  override init() {
+    super.init()
+    loadData()
+  }
+  
+  private func loadData(_ filterString: String? = nil) {
+    accounts = AccountsManager.shared.accounts()
+    accounts.sort { $0.name < $1.name }
+    
+    if let filter = filterString, !filter.isEmpty {
```

Comment:
Do you think we can use case-insensitive search here? Would be easier to find a library when the user doesn't know its exact spelling.

Reply:
Nice catch, I missed that due to the search bar capitalizing the first string by default. Updated